backend/pkg/proto/service.go

```diff
@@ -310,6 +314,15 @@ func (s *Service) decodeConfluentBinaryWrapper(payload []byte) (*confluentEnvelo
 		return nil, fmt.Errorf("failed to read arrLength: %w", err)
 	}
 
+	// Because we can not be sure this is a legit message the read arr length
+	// may be too large to make an array. If that's the case we would get a
+	// panic with: makeslice: len out of range. Therefore, we check this length
+	// before allocating the slice. We assume to not have more than 128 types in
+	// a single message
+	if arrLength > 128 {
+		return nil, fmt.Errorf("arrLength is higher than expected, unlikely a legit envelope")
+	}
+
 	msgTypeIDs := make([]int64, arrLength)
 	// If there is just one msgtype (default index - 0) the array won't be sent at all
 	if arrLength == 0 {
```

Comment:
would there be a chance for `arrLength` to be less than 0 for any reason?

Reply:
Yes, good comment!

Reply:
Fixed, added a check for negative length